command/agent.go

```diff
@@ -700,18 +700,20 @@ func (c *AgentCommand) Run(args []string) int {
 			// Parse 'require_request_header' listener config option, and wrap
 			// the request handler if necessary
 			muxHandler := cacheHandler
-			if lnConfig.RequireRequestHeader {
+			if lnConfig.RequireRequestHeader && !("metrics_only" == lnConfig.Role) {
```

Comment:
Curious why you aren't using `!=` here and below?

Reply:
I think this is my years of Java usage seeping into my tendencies, and you're right, I should change this to be clearer

In Java, the 'correct' way to compare string-not-equals is:
`!"const_string".equals(stringObject)`
You do the const string first, as `stringObject` could be `null`

I'll update this to use `!=` instead, as it's easier at a glance to understand. But it's funny, thinking about the reason I'd made that decision. I didn't even realise, at first.